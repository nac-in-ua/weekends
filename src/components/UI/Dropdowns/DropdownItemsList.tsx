import GeneralPanel from '../../UI/Panels/GeneralPanel'
import DropdownItem from './DropdownItem'

type DropdownItemsListProps = {
  selectedValue: string
  className?: string
  items: Array<string>
  handleSelect: Function
}

function DropdownItemsList({
  items,
  selectedValue,
  handleSelect,
  className,
}: DropdownItemsListProps) {
  return (
    <GeneralPanel className={className}>
      {items.map((item, index) => (
        <DropdownItem
          key={index}
          value={item}
          handleSelect={handleSelect}
          isSelected={selectedValue === item}
        />
      ))}
    </GeneralPanel>
  )
}

export default DropdownItemsList
