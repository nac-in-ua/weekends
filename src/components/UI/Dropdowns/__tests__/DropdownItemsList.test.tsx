import DropdownItemsList from '../DropdownItemsList'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

describe('DropdownItemsList', () => {
  const selectHandler = jest.fn()
  let mockedScrollIntoView = jest.fn()

  beforeAll(() => {
    window.HTMLElement.prototype.scrollIntoView = mockedScrollIntoView
  })

  beforeEach(() => {
    selectHandler.mockClear()
  })

  afterAll(() => {
    mockedScrollIntoView.mockReset()
  })

  it('should render dropdown items list', () => {
    const selectedValue = 'second'
    const { container } = render(
      <DropdownItemsList
        items={['first']}
        handleSelect={selectHandler}
        selectedValue={selectedValue}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render 3 dropdown items', () => {
    const selectedValue = 'second'
    render(
      <DropdownItemsList
        items={['first', 'second', 'third']}
        handleSelect={selectHandler}
        selectedValue={selectedValue}
      />
    )
    expect(screen.getAllByTestId('dropdown-item')).toHaveLength(3)
  })

  it('should handle select', async () => {
    const selectedValue = 'second'
    render(
      <DropdownItemsList
        items={['first', 'second', 'third']}
        handleSelect={selectHandler}
        selectedValue={selectedValue}
      />
    )
    await userEvent.click(screen.getByText('third'))
    expect(selectHandler).toHaveBeenCalledTimes(1)
  })

  it('should handle select with dropdown item value', async () => {
    const selectedValue = 'second'
    render(
      <DropdownItemsList
        items={['first', 'second', 'third']}
        handleSelect={selectHandler}
        selectedValue={selectedValue}
      />
    )
    await userEvent.click(screen.getByText('third'))
    expect(selectHandler).toHaveBeenCalledWith('third')
  })
})
