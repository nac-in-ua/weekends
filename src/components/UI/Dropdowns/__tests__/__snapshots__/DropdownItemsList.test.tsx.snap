// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownItemsList should render dropdown items list 1`] = `
<div>
  <div
    class="absolute overflow-auto rounded-md bg-white shadow-md dark:bg-zinc-900 "
  >
    <div
      class="px-2 py-1 text-sm hover:bg-slate-200 dark:hover:bg-slate-700 font-thin"
      data-testid="dropdown-item"
      data-value="first"
    >
      first
    </div>
  </div>
</div>
`;
